integreat_cms/cms/utils/linkcheck_utils.py

```diff
@@ -55,27 +52,27 @@ def get_urls(
         region = Region.objects.get(slug=region_slug)
         region_links = get_region_links(region)
 
-        if prefetch_content_objects:
-            region_links = region_links.prefetch_related("content_object__language")
         # Prefetch all link objects of the requested region
-        urls = urls.prefetch_related(
-            Prefetch(
-                "links",
-                queryset=region_links,
-                to_attr="region_links",
+        urls = (
+            urls.filter(links__in=region_links)
+            .distinct()
+            .prefetch_related(
+                Prefetch(
+                    "links",
+                    queryset=region_links,
+                    to_attr="region_links",
+                )
             )
         )
-    elif prefetch_content_objects:
-        urls = urls.prefetch_related("links__content_object")
     # Filter out ignored URL types
-    urls = [url for url in urls if url.type not in settings.LINKCHECK_IGNORED_URL_TYPES]
-    if region_slug:
-        # If the region slug is given, only return urls that occur at least once in the requested region
-        urls = [url for url in urls if url.region_links]
+    if settings.LINKCHECK_IGNORED_URL_TYPES:
+        urls = [
+            url for url in urls if url.type not in settings.LINKCHECK_IGNORED_URL_TYPES
+        ]
     return urls
```

Comment:
Returning either the filtered array or the urls variable reduces state mutation and might make the code easier to understand, imo. :)

```suggestion
    if settings.LINKCHECK_IGNORED_URL_TYPES:
        return [url for url in urls if url.type not in settings.LINKCHECK_IGNORED_URL_TYPES]
    else:
        return urls
```